hivemind/dht/schema.py

```diff
@@ -75,19 +76,26 @@ def validate(self, record: DHTRecord) -> bool:
                 return False
             deserialized_record = {key_alias: deserialized_value}
 
-        try:
-            parsed_record = self._schema.parse_obj(deserialized_record)
-        except pydantic.ValidationError as e:
+        parsed_record = None
+        validation_errors = []
+        for schema in self._schemas:
+            try:
+                parsed_record = schema.parse_obj(deserialized_record)
+                break
+            except pydantic.ValidationError as e:
+                validation_errors.append(e)
+        if parsed_record is None:
             readable_record = {self._alias_to_name.get(key_alias, key_alias):
                                deserialized_record[key_alias]}
-            logger.warning(f"Record {readable_record} doesn't match the schema: {e}")
+            logger.warning(
+                f"Record {readable_record} doesn't match the schemas: {validation_errors}")
             return False
 
         parsed_value = parsed_record.dict(by_alias=True)[key_alias]
         if parsed_value != deserialized_record[key_alias]:
             logger.warning(
                 f"Value {deserialized_record[key_alias]} needed type conversions to match "
-                f" the schema: {parsed_value}. Type conversions are not allowed")
+                f"the schema: {parsed_value}. Type conversions are not allowed")
```

Comment:
What if there are two valid schemas, one of which requires conversion and one does not? Is `break` a valid course of action in that case?

Reply:
Fixed this case (it is possible only if two schemas have different annotations for the same key, now we allow both of them).